lib/ringbuf/src/lib.rs

```diff
@@ -183,22 +186,73 @@ macro_rules! ringbuf {
     ($name:ident, $t:ty, $n:expr, $init:expr) => {
         #[allow(dead_code)]
         const _: $t = $init;
+        static $name: () = ();
     };
     ($t:ty, $n:expr, $init:expr) => {
-        #[allow(dead_code)]
-        const _: $t = $init;
+        $crate::ringbuf!(__RINGBUF, $t, $n, $init);
+    };
+}
+
+/// Declares a ringbuffer and set of event counts in the current module or
+/// context.
+///
+/// `counted_ringbuf!(NAME, Type, N, expr)` makes a [`CountedRingbuf`] named `NAME`,
+/// containing entries of type `Type`, with room for `N` such entries, all of
+/// which are initialized to `expr`. `Type` must implement the [`Count`] trait,
+/// which defines how to count occurences of each ringbuf entry variant.
+///
+/// The resulting ringbuffer will be static, so `NAME` should be uppercase. If
+/// you want your ringbuffer to be detected by Humility's automatic scan, its
+/// name should end in `RINGBUF`.
+///
+/// To support the common case of having one quickly-installed ringbuffer per
+/// module, if you omit the name, it will default to `__RINGBUF`.
+///
+#[cfg(not(feature = "disabled"))]
+#[macro_export]
+macro_rules! counted_ringbuf {
+    ($name:ident, $t:ident, $n:expr, $init:expr) => {
+        #[used]
+        static $name: $crate::CountedRingbuf<$t, $n> = $crate::CountedRingbuf {
+            ringbuf: $crate::StaticCell::new($crate::Ringbuf {
+                last: None,
+                buffer: [$crate::RingbufEntry {
+                    line: 0,
+                    generation: 0,
+                    count: 0,
+                    payload: $init,
+                }; $n],
+            }),
+            counters: <$t as $crate::Count>::NEW_COUNTERS,
+        };
+    };
+    ($t:ident, $n:expr, $init:expr) => {
+        $crate::counted_ringbuf!(__RINGBUF, $t, $n, $init);
+    };
+}
+
+#[cfg(feature = "disabled")]
+#[macro_export]
+macro_rules! counted_ringbuf {
+    ($name:ident, $t:ident, $n:expr, $init:expr) => {
+        #[used]
+        static $name: $crate::CountedRingbuf<$t, $n> = $crate::CountedRingbuf {
+            counters: <$t as $crate::Count>::NEW_COUNTERS,
+        };
+    };
+    ($t:ident, $n:expr, $init:expr) => {
+        $crate::counted_ringbuf!(__RINGBUF, $t, $n, $init);
     };
 }
```

Comment:
Note that currently, we still record event counts when the ringbuf crate is compiled with the "disabled" feature flag set. My thinking was that, since `<entry type number of variants> * 4` bytes is generally smaller than a whole ringbuf, where the size of the ringbuf is:
```rust
let entry_sz = size_of::<T>() // payload type
   + 2 // 16-bit line number
   + 2 // 16-bit generation
   + 4 // 32-bit count
   ;
entry_size * N
```
we could still get away with recording the counters on memory-constrained targets.

If others disagree, I'm happy to either change this code so that counts are not recorded when the "disabled" feature is set, or add an additional feature flag to control whether entry counts are recorded.